### ISSUE TITLE
Allow TODO issue with pound sign anywhere in text

### ISSUE THREAD

**Issue by shaunc** [PyCQA/flake8#1916]
### describe the request

Visual studio code prompts to autocreate issues in current repo for todos. It inserts the issue # immediately after colon:
Before
```
# TODO(charlie): fix this
```
After
```
# TODO(charlie): #234 fix this
```
This format seems clear and works with tooltip support in visual studio code.

**Comment by asottile** [comment]
this has nothing to do with flake8

**Comment by shaunc** [comment]
Ah - sorry. I saw that [flake8-todos](https://pypi.org/project/flake8-todos/) didn't have issue tracker and assumed issues must be consolidated here.

**Comment by asottile** [comment]
if you read the issue template or any of the prompts why would that make sense?

**Comment by shaunc** [comment]
Hmm... you mean: "NOTE: flake8 is a linting framework and does not implement any checks"? 

I guess I was not familiar with the architecture, so that note was opaque to me (what is a "check"?). ("All of the individual rules are checked by plugins." or something like that?)

**Comment by asottile** [comment]
this seems pretty clear to me, did you not read this part?

> if you are reporting a problem with a particular check, please track down the plugin which implements that check

**Comment by shaunc** [comment]
That is clear - but it doesn't show up in the template?

![image](https://github.com/PyCQA/flake8/assets/193170/1a9181ff-b8dc-4b6f-a37f-98167a871d36)

Or in the linked "requesting a new feature" in the contribution guide:

![image](https://github.com/PyCQA/flake8/assets/193170/0cb6af9b-56c8-4891-a6eb-ae71b2dadaa5)
